test/unit/player.js

```diff
@@ -152,31 +153,82 @@ test('should asynchronously fire error events during source selection', function
   Lib.log.error.restore();
 });
 
-test('should set the width and height of the player', function(){
-  var player = TestHelpers.makePlayer({ width: 123, height: '100%' });
+// test('should set the width and height of the player', function(){
+//   var player = TestHelpers.makePlayer({ width: 123, height: '100%' });
+//
+//   ok(player.width() === 123);
+//   ok(player.el().style.width === '123px');
+//
+//   var fixture = document.getElementById('qunit-fixture');
+//   var container = document.createElement('div');
+//   fixture.appendChild(container);
+//
+//   // Player container needs to have height in order to have height
+//   // Don't want to mess with the fixture itself
+//   container.appendChild(player.el());
+//   container.style.height = '1000px';
+//   ok(player.height() === 1000);
+//
+//   player.dispose();
+// });
 
-  ok(player.width() === 123);
-  ok(player.el().style.width === '123px');
+test('should set the width, height, and aspect ratio via a css class', function(){
+  let player = TestHelpers.makePlayer();
+  let getStyleText = function(styleEl){
+    return (styleEl.styleSheet && styleEl.styleSheet.cssText) || styleEl.innerHTML;
+  };
 
-  var fixture = document.getElementById('qunit-fixture');
-  var container = document.createElement('div');
-  fixture.appendChild(container);
+  ok(player.styleEl_.parentNode === player.el(), 'player has a style element');
+  ok(!getStyleText(player.styleEl_), 'style element should be empty when the player is given no dimensions');
 
-  // Player container needs to have height in order to have height
-  // Don't want to mess with the fixture itself
-  container.appendChild(player.el());
-  container.style.height = '1000px';
-  ok(player.height() === 1000);
+  let rules;
 
-  player.dispose();
-});
+  function getStyleRules(){
```

Comment:
It would be cool if we could use this CSS AST other places, but right now this is the only place we add styles by creating a new style element. Other places we could only check if a class was added appropriately.